### PR TITLE
Issue 130 !sch handle all day events

### LINKED ISSUE #130: `!sch` command doesn't correctly handle all-day events
- Consider updating googleapis, which could make implementing these changes easier. The docs for the version of the library currently in use [404s](https://googleapis.dev/nodejs/googleapis/61.0.0).
- Convert start time of all-day events to broadcaster's timezone (as configured in bot). Currently, the start time is in UTC.
- Detect when it's during an all-day event and convey that to the user.
- If it's during an all-day event and there is a scheduled event during the all-day event, tell the user when it begins.

### PR DESCRIPTION
**Pull request by Riott**
Updated googleapis to 118.0.0
Use the time zone of the broadcast configuration for the start time of all day events
Convey all day events to the user,
Convey the next coming event within an all day event.
Convey any events that are currently happening.

image

resolves #130

### CODE REVIEW

File: lib/services/schedule.js [hunk 1]
```diff
@@ -13,26 +14,34 @@ class GoogleCal {
...
-          moment.tz(event.start.dateTime, 'YYYY-MM-DDTHH:mm:ss', event.start.timeZone).isBefore(now)
-        );
       });
-
       if (!nextEvent) {
         return null;
       }
-
+      
+      const startTime = moment.tz(nextEvent.start.dateTime, 'YYYY-MM-DDTHH:mm:ss', this.timezone)
```
Comment (11k) [inline lib/services/schedule.js:26]: The issue does say to use the broadcaster's timezone as configured in the bot, but I no longer think that's the right call. The broadcaster's timezone and the timezone of their calendar might differ. If that's the case, these dates won't be parsed correctly. For instance, Destiny lives in `America/New_York`, but his calendar is in `America/Chicago`.

Let's revert to using the calendar's timezone. It's a little annoying that we only get a timezone for non-all-day events, however. I was originally going to suggest you use [this API](https://developers.google.com/calendar/api/v3/reference/calendars/get), but that requires authorization. Considering the calendar timezone will change rarely (if at all), maybe an easier approach is to introduce a new config named `calendarTimezone`.

File: lib/services/schedule.js [hunk 2]
```diff
@@ -13,26 +14,34 @@ class GoogleCal {
 
   findNextStreamDay() {
     return this.getListOfUpcomingEvents(this.configuration.GOOGLE_CALENDAR_ID).then((events) => {
-      const now = moment.tz('UTC');
       const nextEvent = events.find((event) => {
```
Comment (11k) [inline lib/services/schedule.js:17]: You're just trying to get the next event and return `null` if there isn't one? Why not immediately return `null` if `events` is empty? I don't think the response will ever contain an event without a start, so you're already doing this, but in a convoluted way.

File: lib/services/schedule.js (hunk 1 above)
Comment (11k) [reply lib/services/schedule.js:26]: I just realized that [this API](https://developers.google.com/calendar/api/v3/reference/events/list) (the one we use for events) accepts a timezone as a parameter. Using this parameter probably the best solution. You can just get all the dates/times in UTC and that would remove timezones from the equation altogether.

**Review by 11k — CHANGES_REQUESTED** [review]

File: lib/services/schedule.js (hunk 1 above)
Comment (Riott) [reply lib/services/schedule.js]: I've added the timezone parameter, then used .utc in the schedule service and then just regular moments in the command implementation I believe this is the intent right? so we grab it as a utc date from the calendar then convert to the local timezone.

**Review by 11k — APPROVED** [review]

**Pull request merged**